suite-native/storage/src/migrations/walletSettings/v1.ts

```diff
@@ -1,36 +1,123 @@
+import { initMmkvStorage } from '../../storage';
 
-const migrateEnabledDiscoveryNetworkSymbols = (
-    oldEnabledDiscoveryNetworkSymbols: NetworkSymbol[],
-): NetworkSymbol[] =>
-    (oldEnabledDiscoveryNetworkSymbols as NetworkSymbolOld[]).map(networkSymbol =>
-        networkSymbol === 'bnb' ? 'bsc' : networkSymbol,
-    );
+const getFiatCurrencyCode = (appSettings: object): string | undefined => {
+    if ('fiatCurrencyCode' in appSettings && typeof appSettings.fiatCurrencyCode === 'string') {
+        return appSettings.fiatCurrencyCode;
+    }
+
+    if (
+        'fiatCurrency' in appSettings &&
+        typeof appSettings.fiatCurrency === 'object' &&
+        appSettings.fiatCurrency !== null &&
+        'label' in appSettings.fiatCurrency &&
+        typeof appSettings.fiatCurrency.label === 'string'
+    ) {
+        return appSettings.fiatCurrency.label;
+    }
+
+    return undefined;
+};
+
+const migrateLocalCurrency = (appSettings: object): BaseCurrencyCode | undefined => {
+    const localCurrency = getFiatCurrencyCode(appSettings);
+
+    if (!localCurrency || !isBaseCurrencyCode(localCurrency)) return undefined;
 
-// @ts-expect-error
-const deprecatedNetworks: NetworkSymbol[] = ['dash', 'btg', 'nmc', 'vtc', 'dgb'];
+    return localCurrency;
+};
 
-const migrateDiscoveryDeprecateNetworks = (
-    oldEnabledDiscoveryNetworkSymbols: NetworkSymbol[],
-): NetworkSymbol[] =>
-    oldEnabledDiscoveryNetworkSymbols.filter(
-        networkSymbol => !deprecatedNetworks.includes(networkSymbol),
+const migrateBitcoinAmountUnit = (appSettings: object): PROTO.AmountUnit | undefined => {
+    if (!('bitcoinUnits' in appSettings)) return undefined;
+
+    const bitcoinAmountUnit = appSettings.bitcoinUnits;
+
+    if (
+        typeof bitcoinAmountUnit !== 'number' ||
+        // only BTC or SATS are supported
+        ![PROTO.AmountUnit.BITCOIN, PROTO.AmountUnit.SATOSHI].includes(bitcoinAmountUnit)
+    ) {
+        return undefined;
+    }
+
+    return bitcoinAmountUnit;
+};
+
+const migrateBscNetworkSymbol = (oldEnabledDiscoveryNetworkSymbols: string[]): string[] =>
+    oldEnabledDiscoveryNetworkSymbols.map(networkSymbol =>
+        networkSymbol === 'bnb' ? 'bsc' : networkSymbol,
     );
 
+const filterUnknownNetworkSymbols = (networkSymbols: string[]): NetworkSymbol[] =>
+    networkSymbols.filter(networkSymbol => isNetworkSymbol(networkSymbol));
+
 /**
  * Migration of discoveryConfig slice, which was declared locally in suite-native,
  * to the walletSettings reducer shared in suite-common.
  * All migrations that were done on discoveryConfig are moved here
  */
-export const migrateDiscoveryConfigToWalletSettings = (
-    oldEnabledDiscoveryNetworkSymbols: NetworkSymbol[],
-): NetworkSymbol[] =>
-    pipe(
-        oldEnabledDiscoveryNetworkSymbols,
-        migrateEnabledDiscoveryNetworkSymbols,
-        migrateDiscoveryDeprecateNetworks,
+const migrateDiscoveryConfigToWalletSettings = <T extends object>(
+    discoveryConfig: T,
```

Comment:
Generic is useless. I guess that originally you wanted to do something more with it, but no longer relevant?
```suggestion
    discoveryConfig: object,
```

Reply:
b7264f9d2aca80d2a93e69c7409c301a2cd78a16